app/controllers/dashboard/dashboard-activity-controller.js

```diff
@@ -9,29 +9,49 @@ const moment = require('moment-timezone')
 const response = require('../../utils/response').response
 const CORRELATION_HEADER = require('../../utils/correlation_header').CORRELATION_HEADER
 const ConnectorClient = require('../../services/clients/connector_client').ConnectorClient
-const { isADirectDebitAccount } = require('../../services/clients/direct_debit_connector_client.js')
+const {isADirectDebitAccount} = require('../../services/clients/direct_debit_connector_client.js')
 const auth = require('../../services/auth_service.js')
 const connectorClient = () => new ConnectorClient(process.env.CONNECTOR_URL)
 const datetime = require('../../utils/nunjucks-filters/datetime')
 
 module.exports = (req, res) => {
   const correlationId = _.get(req, 'headers.' + CORRELATION_HEADER, '')
   const gatewayAccountId = auth.getCurrentGatewayAccountId((req))
+  let fromDateTime
   let toDateTime = moment().tz('Europe/London').format() // Today is the default
   let daysAgo = 0
-  let period = _.get(req, 'query.period', 'today')
+  const period = _.get(req, 'query.period', 'today')
 
   if (period === 'yesterday') {
     daysAgo = 1
   } else if (period === 'previous-seven-days') {
     daysAgo = 8 // 7+1 because we count starting from yesterday
   } else if (period === 'previous-thirty-days') {
     daysAgo = 31 // 30+1 because we count starting from yesterday
+  } else if (period === 'custom') {
+    fromDateTime = _.get(req, 'query.fromDateTime', null)
+    toDateTime = _.get(req, 'query.toDateTime', null)
+    if (!fromDateTime || !toDateTime || !moment(fromDateTime).isValid() || !moment(toDateTime).isValid()) {
```

Comment:
when we discussed this I thought we said we keep it so you can only specify one end of the window, rather than the range, as this opens us up to DB spamming?

Reply:
Also I think it makes things more readable if you alias long chains of booleans with what they represent eg
```
validFromDate = fromDateTime &&  moment.isValid(fromDateTime)
```